File: src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.java
package com.hedvig.claims.commands;

import com.hedvig.claims.aggregates.ClaimSource;
import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

import java.time.Instant;
import java.time.LocalDateTime;

@Value
public class CreateBackofficeClaimCommand {
  @TargetAggregateIdentifier
  String id;
  String memberId;
  Instant registrationDate;
  ClaimSource claimSource;
}

File: src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.kt
package com.hedvig.claims.commands

import com.hedvig.claims.aggregates.ClaimSource
import lombok.Value
import org.axonframework.commandhandling.TargetAggregateIdentifier
import java.time.Instant
import java.util.UUID

class CreateBackofficeClaimCommand(
    @TargetAggregateIdentifier
    val id: String,
    val memberId: String,
    val registrationDate: Instant,
    val claimSource: ClaimSource,
    val contractId: UUID?
)
